src/identity/manager.rs

```diff
@@ -184,12 +181,9 @@ impl<T: CertificateProvider + Clone + Send + 'static> CertificateProvider for Se
                             let (tx, rx) = watch::channel(None);
                             write_locked_cache.insert(id.clone(), rx.clone());
                             drop(write_locked_cache);
-                            tokio::spawn(SecretManager::refresh_handler(
-                                id.clone(),
-                                self.cache.clone(),
-                                self.client.clone(),
-                                tx,
-                            ));
+                            let s = self.clone();
```

Comment:
Doesn't this clone bring much cost

Reply:
No, it is designed to be cheap to clone (I also added a comment in the code to clarify this). It is just incrementing a reference counter

Reply:
Also what tonic recommends

Reply:
Client is also an Arc under the hood and designed (by tonic) to be cheap to clone

Reply:
Thanks for clarify